rollup.config.js

```diff
@@ -0,0 +1,123 @@
+/* eslint-env node */
+
+import commonJs from '@rollup/plugin-commonjs';
+import typescript from '@rollup/plugin-typescript';
+import { nodeResolve } from '@rollup/plugin-node-resolve';
+import replace from '@rollup/plugin-replace';
+import { terser } from 'rollup-plugin-terser';
+import cleanup from 'rollup-plugin-cleanup';
+import pkg from './package.json';
+
+const input = 'src/index.ts';
+const external = Object.keys( pkg.peerDependencies || {} );
+const banner = `/**
+* @license Copyright (c) 2003-2021, CKSource - Frederico Knabben. All rights reserved.
+* For licensing, see LICENSE.md.
+*/`;
+
+export default [
+	// Creates `umd` build that can be directly consumed via <script /> tag (development mode).
+	// Setting `NODE_ENV` value to `development` enables dev utils such as prop-types.
+	{
+		input,
+		external,
+		output: {
+			banner,
+			format: 'umd',
+			file: 'dist/index.umd.development.js',
+			name: 'CKEditor4React',
+			globals: {
+				react: 'React'
+			}
+		},
+		plugins: [
+			typescript(),
+			nodeResolve(),
+			commonJs(),
+			replace( {
+				preventAssignment: true,
+				values: {
+					'process.env.NODE_ENV': '"development"'
+				}
+			} ),
+			cleanupPlugin()
+		]
+	},
+	// Creates `umd` build that can be directly consumed via <script /> tag (production mode).
+	// Setting `NODE_ENV` value to `production` disables dev utils.
+	{
+		input,
+		external,
+		output: {
+			banner,
+			format: 'umd',
+			file: 'dist/index.umd.production.min.js',
+			name: 'CKEditor4React',
+			globals: {
+				react: 'React'
+			}
+		},
+		plugins: [
+			typescript(),
+			nodeResolve(),
+			commonJs(),
+			replace( {
+				preventAssignment: true,
+				values: {
+					'process.env.NODE_ENV': '"production"'
+				}
+			} ),
+			cleanupPlugin(),
+			terser()
+		]
+	},
+	// Creates `cjs` build that can be further optimized downstream.
+	{
+		input,
+		external: external.concat( [
+			'ckeditor4-integrations-common',
+			'prop-types'
+		] ),
+		output: {
+			banner,
+			format: 'cjs',
+			file: 'dist/index.cjs.js'
+		},
+		plugins: [ typescript(), cleanupPlugin() ]
```

Comment:
After some considerations, I think it's better to use default `es5` target in both cases. It's not fair for developers targeting older browsers to transpile our library. For example, if they used webpack's `babel-loader` they would need to hack around `exclude: /node_modules/` to get our library transpiled. I will also get rid of `Object.entries` and `Array.prototype.includes` so that polyfills are not required for them.

On a side note, including polyfills in the library is a different story. I'd rather not do that. I will list required IE11 polyfills in the README though (it should be just `promise` I believe).